src/config-validator.ts

```diff
@@ -67,15 +67,16 @@ export const getArray = (name: string): ZodString['_output'][] => {
 	}
 };
 
-export const getEnum = <T extends [string, ...string[]]>(name: string, zEnum: ZodEnum<T>): z.infer<ZodEnum<T>> => {
-    const value = process.env[name];
+export const getEnum = <T extends [string, ...string[]]>(name: string, zEnum: ZodEnum<T>, value: string | undefined = ''): z.infer<ZodEnum<T>> => {
```

Comment:
'value' is not the default but rather the opposite of it.  It will take priority over the env variable and be ignored only when it is blank/empty/not present. The value for 'value' is either empty for `AUTH_STRATEGY` or is one of the items in the `SECURED_API` list and is validated to have values from z.enums belonging to both i.e one of the allowed values of T and that's a string (not a z.enum). And so making 'value' to be of type z.infer<ZodEnum<T>> will not work.

```
export const apiList = ['CREATE', 'FIND'] as const;
export const authList = ['EGO', 'KEYCLOAK', 'NONE'] as const;
```

```
export const securedApi = getArray('SECURED_API');
securedApi.forEach(api => {getEnum('SECURED_API', z.enum(apiList), api)});

export const authStrategy: AuthStrategy = getEnum('AUTH_STRATEGY', z.enum(authList));
```

```
export const getEnum = <T extends [string, ...string[]]>(name: string, zEnum: ZodEnum<T>, value: string | undefined = ''): z.infer<ZodEnum<T>> => {
    const stringValue = zEnum.safeParse(value || process.env[name]);
    if (!stringValue.success) {
        throw new Error(`Environment variable ${name} has an invalid value. Valid values are: ${zEnum.options}`);
    }
    return stringValue.data;
};
```

Reply:
It feels like we are overloading this function then. The function was originally validating if an environment variable was an enum. With this `value` parameter it is now validating that value and ignoring the environment value... this second behaviour we don't need this function for since we already have validation for enums through our zod schema.

It will be easier for use to reason about these two behaviours if they are not combined into a single function.